BoxContentSDK/BoxContentSDK/OAuth2/BOXOAuth2Session.m

```diff
@@ -64,14 +64,17 @@ - (void)performAuthorizationCodeGrantWithReceivedURL:(NSURL *)URL withCompletion
     NSString *authorizationCode = [URLQueryParams valueForKey:BOXAuthURLParameterAuthorizationCodeKey];
     NSString *authorizationError = [URLQueryParams valueForKey:BOXAuthURLParameterErrorCodeKey];
 
-    if ((_nonce || serverNonce) && [serverNonce isEqualToString:_nonce] == NO) {
-        NSError *error = [[NSError alloc] initWithDomain:BOXContentSDKErrorDomain 
-                                                    code:BOXContentSDKAuthErrorAccessTokenNonceMismatch
-                                                userInfo:nil];
-        if (block) {
-            block(nil, error);
+    NSString *authenticationRedirectURIScheme = [[NSURL URLWithString:self.redirectURIString] scheme];
```

Comment:
From what I checked with the server team, we are not 100% confident that the server will always return a state param(when it is empty) so we cannot base it on that. We will go with the current fix.